code/game/machinery/autodoc_excelsior.dm

```diff
@@ -0,0 +1,249 @@
+		to_chat(user, SPAN_WARNING("Subject cannot have abiotic items on."))
+		return
+	if (target.buckled)
+		to_chat(user, SPAN_NOTICE("Unbuckle the subject before attempting to move them."))
+		return
+	user.visible_message(
+		SPAN_NOTICE("\The [user] begins placing \the [target] into \the [src]."),
+		SPAN_NOTICE("You start placing \the [target] into \the [src].")
+	)
+	if(!do_after(user, 30, src) || !Adjacent(target))
+		return
+	set_occupant(target)
+	src.add_fingerprint(user)
+	return
+
+/obj/machinery/excelsior_autodoc/Process()
+	if(stat & (NOPOWER|BROKEN))
+		autodoc_processor.stop()
+		update_icon()
+
+/obj/machinery/excelsior_autodoc/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = NANOUI_FORCE_OPEN, var/datum/topic_state/state = GLOB.default_state)
+	autodoc_processor.ui_interact(user, ui_key, ui, force_open, state)
+
+/obj/machinery/excelsior_autodoc/Topic(href, href_list)
+	return autodoc_processor.Topic(href, href_list)
+
+/obj/machinery/excelsior_autodoc/proc/open_cover()
+	cover_state = image(icon, "opening")
+	cover_state.layer = 5
+	cover_closed = FALSE
+	cover_moving = TRUE
+	update_icon()
+
+/obj/machinery/excelsior_autodoc/proc/close_cover()
+	cover_state = image(icon, "closing")
+	cover_state.layer = 5
+	cover_closed = TRUE
+	cover_moving = TRUE
+	update_icon()
+
+/obj/machinery/excelsior_autodoc/on_update_icon()
+
+	cut_overlays()
+
+	if(panel_open)
+		var/image/panel = image(icon, "panel")
+		panel.layer = 4.5
+		add_overlays(panel)
+
+	if(occupant)
+		var/image/comrade = image(occupant.icon, occupant.icon_state)
+		comrade.overlays = occupant.overlays
+		comrade.pixel_x = 6
+		comrade.layer = 4
+		add_overlays(comrade)
+		add_overlays(cover_state)
+		if(cover_moving)
+			sleep (15)
+			if(cover_closed)
+				cover_state = image(icon, "closed")
+				cover_state.layer = 4.5
+				cover_moving = FALSE
+				update_icon()
+			else
+				cover_state = image(icon, "opened")
+				cover_state.layer = 4.5
+				cover_moving = FALSE
+				update_icon()
+
+		var/actual_health = occupant.maxHealth - (occupant.getBruteLoss() + occupant.getFireLoss() + occupant.getOxyLoss() + occupant.getToxLoss())
+		if(actual_health < 1)
+			screen_state = image(icon, "screen_00")
+		else if(actual_health < 10)
+			screen_state = image(icon, "screen_10")
+		else if(actual_health < 20)
+			screen_state = image(icon, "screen_20")
+		else if(actual_health < 30)
+			screen_state = image(icon, "screen_30")
+		else if(actual_health < 40)
+			screen_state = image(icon, "screen_40")
+		else if(actual_health < 50)
+			screen_state = image(icon, "screen_50")
+		else if(actual_health < 60)
+			screen_state = image(icon, "screen_60")
+		else if(actual_health < 70)
+			screen_state = image(icon, "screen_70")
+		else if(actual_health < 80)
+			screen_state = image(icon, "screen_80")
+		else if(actual_health < 90)
+			screen_state = image(icon, "screen_90")
+		else if(actual_health <= 100)
+			screen_state = image(icon, "screen_100")
+
```

Comment:
consider dividing by 10 and using clamp/round/ceiling/floor instead of a huge else if chain.

Reply:
That would take up even more space with no gain whatsoever.

Reply:
no it wouldn't... it  should take 5 lines in worst case..

Reply:
`screen_state = image(icon, "screen_[round((actual_health/maxHealth)*100, 10)]")`
I believe this would work, and would also work if max health was something other than 100.

Reply:
No it wouldn't.

Reply:
right, I forgot my foray into sprite-work.
iirc it needs a +, so it's more of...
`var/icontobe = screen_`
`icontobe += "[round((actual_health/maxHealth)*100, 10)]"`
`screen_state = image(icon, icontobe)`

Reply:
okay, I'm going to start putting more effort into this, this is my third time messing up in just this...

Reply:
It doesn't work either. Also it's `occupant.maxHealth`.

Reply:
> 
> 
> Doesn't work either. Also it's `occupant.maxHealth`.

that's why I deleted the false correction and edited in occupant.maxHealth.